### PR TITLE
feat: Added support for API destinations

### PR DESCRIPTION
**Pull request by svenlito**
- feat: support api destination

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -246,7 +246,7 @@ module "eventbridge" {
 
 | Name | Version |
 |------|---------|
-| <a name="provider_aws"></a> [aws](#provider\_aws) | >= 3.40 |
+| <a name="provider_aws"></a> [aws](#provider\_aws) | 3.51.0 |
```
Comment (antonbabenko) [inline README.md:249]: Please use `terraform-docs` 0.13.0 for the time being to generate proper docs.

File: examples/with-api-destination/.tool-versions [hunk 2]
```diff
@@ -0,0 +1 @@
+terraform 0.15.0
```
Comment (antonbabenko) [inline examples/with-api-destination/.tool-versions:1]: Please remove this file.

File: examples/with-api-destination/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# EventBridge API Destination Example
+
+Configuration in this directory creates resources to control access to EventBridge.
```
Comment (antonbabenko) [inline examples/with-api-destination/README.md:3]:
```suggestion
Configuration in this directory creates resources to control access to EventBridge using API destinations.
```

Or something like that.

File: examples/with-api-destination/main.tf [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          value = random_pet.this.id
+        }
+      }
+    }
+  }
+
+  api_destinations = {
+    "requestbin" = {
+      description                      = "my requestbin endpoint"
+      invocation_endpoint              = "https://pipedream.com/@svenlito/test-rb-1-p_rvCQGl1"
```
Comment (antonbabenko) [inline examples/with-api-destination/main.tf:45]: Let's use something publicly available. I usually use this service - https://smee.io/

File: examples/with-api-destination/main.tf [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+##################
+# Extra resources
+##################
+
+resource "random_pet" "this" {
+  length = 2
+}
+
+resource "aws_cloudwatch_event_bus" "existing_bus" {
```
Comment (antonbabenko) [inline examples/with-api-destination/main.tf:66]: This resource does not seem to be used.

**Review by antonbabenko — COMMENTED** [review]
Looks good, but I don't have an answer on `too many elements in dynamic block` issue you described. Maybe it is an open bug in the provider?

**Comment by svenlito** [comment]
> Looks good, but I don't have an answer on `too many elements in dynamic block` issue you described. Maybe it is an open bug in the provider?

Saw something once, will dig it out again. Thanks for the feedback, will tidy up the remainder. 👍🏻

**Review by antonbabenko — APPROVED** [review]
Looks good. Are there outstanding issues/bugs related to the added functionality? Or is it ready to be merged right away?

**Comment by svenlito** [comment]
> Looks good. Are there outstanding issues/bugs related to the added functionality? Or is it ready to be merged right away?

I've noted in the readme that there is missing functionality but fwiw it's ready to use. I might want to add a more complete example however. Currently it's not really documenting how to map targets/rules to the API destinations.

**Comment by antonbabenko** [comment]
I think we need to have that documented before we merge this one. It is really important :)

**Comment by svenlito** [comment]
> It is really important :)
Thought it was :)

**Comment by svenlito** [comment]
@antonbabenko funny enough it seems to be completely undocumented.

**Comment by svenlito** [comment]
ok, got it. https://github.com/hashicorp/terraform-provider-aws/issues/20343
will have to rework the PR

**Comment by svenlito** [comment]
sorted now.

File: main.tf [hunk 6]
```diff
@@ -238,6 +238,116 @@ resource "aws_cloudwatch_event_connection" "this" {
...
+
+          dynamic "oauth_http_parameters" {
+            for_each = lookup(each.value.auth_parameters.oauth, "oauth_http_parameters", null) != null ? [
+              each.value.auth_parameters.oauth.oauth_http_parameters
+            ] : []
+
+            content {
+              dynamic "body" {
+                for_each = lookup(each.value.auth_parameters.oauth.oauth_http_parameters, "body", null) != null ? [
+                  each.value.auth_parameters.oauth.oauth_http_parameters.body
```
Comment (svenlito) [inline main.tf:270]: @antonbabenko could use your help or suggestion here. `body` as well as `header` and `query_string` can occur multiple times. Currently things get override by the last iteration. I was thinking `for b in value : b` would sort it out but then I'm ending up with an unusable value. Perhaps you have an idea?

Comment (antonbabenko) [reply main.tf:270]: I will take a look later today or tomorrow. I am too busy with other things this week.

**Review by antonbabenko — APPROVED** [review]

File: main.tf (hunk 6 above)
Comment (antonbabenko) [reply main.tf]: The magic here is to require a `list of maps` and iterate over them (in S3 and CloudFront modules I did a similar solution, but for 0.13+ the code can be a bit simpler in some cases). I have verified this PR altogether, and it works.

**Comment by antonbabenko** [comment]
Thank you @svenlito !

[v1.13.0](https://github.com/terraform-aws-modules/terraform-aws-eventbridge/releases/v1.13.0) has been just released.

File: main.tf (hunk 6 above)
Comment (svenlito) [reply main.tf]: @antonbabenko appreciate the help!

**Pull request merged**